src/main/scala/org/apache/spark/scheduler/cluster/armada/ArmadaClusterManagerBackend.scala

```diff
@@ -73,6 +73,7 @@ private[spark] class ArmadaClusterSchedulerBackend(
       EnvVar().withName("SPARK_EXECUTOR_CORES").withValue("1"),
       EnvVar().withName("SPARK_EXECUTOR_MEMORY").withValue("512m"),
       EnvVar().withName("SPARK_DRIVER_URL").withValue(driverURL),
+      EnvVar().withName("SPARK_JAVA_OPT_1").withValue("--add-opens=java.base/sun.nio.ch=ALL-UNNAMED"),
```

Comment:
what is this?

Reply:
When Spark moved to Java 17, running Spark an Spark tests fails with

    Caused by: java.lang.IllegalAccessError: class org.apache.spark.storage.StorageUtils$ (in unnamed module @0x59d016c9) cannot access class sun.nio.ch.DirectBuffer (in module java.base) because module java.base does not export sun.nio.ch to unnamed module @0x59d016c9

if this and many other `--add-opens` are missing from the java command line:
https://github.com/apache/spark/blob/ef4cd39a03136cee2a3d6dff24cf5f893f69822a/pom.xml#L312-L332

https://stackoverflow.com/questions/73465937/apache-spark-3-3-0-breaks-on-java-17-with-cannot-access-class-sun-nio-ch-direct